cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+            cluster_case_name = columbia_case_name
+    else:
+        # We don't care if data is empty or both are empty
+        pass
+
+    if cluster_case_name_full and cluster_case_name:
+        if len(cluster_case_name) > len(cluster_case_name_full):
+            # Swap field values
+            update_dict["case_name"] = cluster_case_name_full
+            update_dict["case_name_full"] = cluster_case_name
+
+    return update_dict
+
+
+def merge_date_filed(
+    cluster: OpinionCluster, columbia_data: dict
+) -> dict[str, Any]:
+    """Merge date filed
+    :param cluster: The cluster of the merging item
+    :param columbia_data: json data from columbia
+    """
+
+    columbia_date_filed = columbia_data.get("date_filed")
+    cluster_date_filed = cluster.date_filed
+
+    if columbia_date_filed:
+        if cluster.docket.source == Docket.SCRAPER:
+            # Give preference to columbia data
+            if columbia_date_filed != cluster_date_filed:
+                return {"date_filed": columbia_date_filed}
+
+    return {}
+
+
+def merge_long_fields(
+    field_name: str,
+    overlapping_data: Optional[tuple[str, str]],
+    cluster_id: int,
+) -> dict[str, Any]:
+    """Merge two long text fields
+    :param field_name: Field name to update in opinion cluster
+    :param overlapping_data: data to compare from columbia and courtlistener
+    :param cluster_id: cluster id
+    """
+    if not overlapping_data:
+        return {}
+
+    columbia_data, cl_data = overlapping_data
+    # Do some text comparison
+    similarity = get_cosine_similarity(columbia_data, cl_data)
+    if similarity < 0.9:
+        # they are not too similar, choose the larger one
+        if len(columbia_data) > len(cl_data):
+            return {field_name: columbia_data}
+
+    else:
+        if similarity <= 0.5:
+            logger.info(
+                f"The content compared is very different. Cluster id: {cluster_id}"
+            )
+    return {}
+
+
+def merge_strings(
+    field_name: str, overlapping_data: tuple[str, str]
+) -> dict[str, Any]:
+    """Compare two strings and choose the largest
+    :param field_name: field name to update in opinion cluster
+    :param overlapping_data: data to compare from columbia and courtlistener
+    """
+    if not overlapping_data:
+        return {}
+
+    columbia_data, cl_data = overlapping_data
+    if len(columbia_data) > len(cl_data):
+        return {field_name: columbia_data}
+
+    return {}
+
+
+def merge_judges(
+    overlapping_data: Optional[tuple[str, str]],
+) -> dict[str, Any]:
+    """Merge overlapping judge values
+    :param overlapping_data: data to compare from columbia and courtlistener
+    """
+
+    if not overlapping_data:
+        return {}
+
+    columbia_data, cl_data = overlapping_data
+    # We check if any word in the string is uppercase
+    cl_data_upper = (
+        True if [s for s in cl_data.split(",") if s.isupper()] else False
+    )
+
```

Comment:
is the same code we used for harvard?

Reply:
yes, i moved to corpus_importer/utils.py so that it can be reused with harvard/columbia